src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -887,5 +887,196 @@ public static void CheckLegacyNullRowVersionIsEmptyArray()
                 Assert.Equal(result, reader.GetFieldValue<byte[]>(0));
             }
         }
+
+
+        private class _Row
+        {
+            public int Id;
+            public Guid DocumentIdentificationId;
+            public string Name;
+            public string Value;
+
+            public _Row(int id, Guid documentIdentificationId, string name, string value)
+            {
+                Id = id;
+                DocumentIdentificationId = documentIdentificationId;
+                Name = name;
+                Value = value;
+            }
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async Task CanReadAwkwardDataLengths()
```

Comment:
You seem to have modified 3 paths (TryReadStringWithEncoding, TryReadByteArrayWithContinue, and TryReadPlpUnicodeCharsWithContinue) where this issue can occur - can you add test coverage for all of the changes?

Reply:
All the paths already have coverage. The code changes retain the existing functionality. Proving they were susceptible to the problem found with chars specifically would require that I write some sort of fuzzer to generate large amounts of varied input data and make sure it can be read correctly. It's an idea that I've considered but is not sensible to do for this PR because of the need to quickly fix it, it can cause silent bad data to be returned to users, highly dangerous.